src/System.Runtime.Environment/src/OSName.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Runtime.InteropServices
+{
+    public struct OSName : IEquatable<OSName>
+    {
+        private readonly string _name;
+
+        private const string WindowsName = "WINDOWS";
+        private const string LinuxName = "LINUX";
+        private const string OSXName = "OSX";
+
+        public OSName(string name)
+        {
+            _name = name;
```

Comment:
Why is this valuable?

Sent from my Windows Phone
________________________________
From: Lakshmi Priya<mailto:notifications@github.com>
Sent: ‎4/‎24/‎2015 10:50 AM
To: dotnet/corefx<mailto:corefx@noreply.github.com>
Cc: David Kean<mailto:David.Kean@microsoft.com>
Subject: Re: [corefx] [Api-addition] System.Runtime.Environment - support for platform detection. (#1494)


In src/System.Runtime.Environment/src/OSName.cs<https://github.com/dotnet/corefx/pull/1494#discussion_r29069795>:

> +?// Copyright (c) Microsoft. All rights reserved.
> +// Licensed under the MIT license. See LICENSE file in the project root for full license information.
> +
> +namespace System.Runtime.InteropServices
> +{
> +    public struct OSName : IEquatable<OSName>
> +    {
> +        private readonly string _name;
> +
> +        private const string WindowsName = "WINDOWS";
> +        private const string LinuxName = "LINUX";
> +        private const string OSXName = "OSX";
> +
> +        public OSName(string name)
> +        {
> +            _name = name;


The _name can have null and empty values, we are not throwing for those.

—
Reply to this email directly or view it on GitHub<https://github.com/dotnet/corefx/pull/1494/files#r29069795>.